src/main/groovy/wooga/gradle/unity/tasks/ExecuteCsharpScript.groovy

```diff
@@ -0,0 +1,82 @@
+package wooga.gradle.unity.tasks
+
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import wooga.gradle.unity.UnityTask
+
+import java.nio.file.Files
+
+class ExecuteCsharpScript extends UnityTask {
+
+    private final RegularFileProperty sourceScript = objects.fileProperty()
+
+    @InputFile
+    RegularFileProperty getSourceScript() {
+        return sourceScript
+    }
+
+    void setSourceScript(Provider<RegularFile> sourceCsScript) {
+        this.sourceScript.set(sourceCsScript)
+    }
+
+    void setSourceScript(RegularFile sourceCsScript) {
+        this.sourceScript.set(sourceCsScript)
+    }
+
+    void setSourceScript(File sourceCsScript) {
+        this.sourceScript.set(sourceCsScript)
+    }
+
+    private final RegularFileProperty destScript = objects.fileProperty()
+
+    @InputFile
+    RegularFileProperty getDestScript() {
+        return destScript

```

Comment:
Since `sourceScript` has a full name, I believe this one should. `destinationScript`?